lib/Controller/SAMLController.php

```diff
@@ -85,13 +85,34 @@ public function __construct($appName,
 	 */
 	private function autoprovisionIfPossible(array $auth) {
 		$uidMapping = $this->config->getAppValue('user_saml', 'general-uid_mapping');
+		$displaynamefirstpartMapping = $this->config->getAppValue('user_saml','general-displaynamefirstpart_mapping');
+		$displaynamesecondpartMapping = $this->config->getAppValue('user_saml','general-displaynamesecondpart_mapping');
```

Comment:
Displayname works for me everywhere: upper right side, personal page, search field for sharing items.

regarding mailaddress: i have not found time to implement this better.

My current solution is: writing the email the moment the user gets created, no update, no changes:

SAMLController.php
```
$mailMapping = "urn:oid:0.9.2342.19200300.100.1.3";

                $mail='';
                if(isset($auth[$mailMapping])) {
                        if(is_array($auth[$mailMapping])) {
                                $mail = $auth[$mailMapping][0];
                        } else {
                                $mail = $auth[$mailMapping];
                        }
                }`

```

```
if(!$userExists && !$autoProvisioningAllowed) {
                                throw new NoUserFoundException();
                        } elseif(!$userExists && $autoProvisioningAllowed) {
                                $this->userBackend->createUserIfNotExists($uid,$displayName);
                                $user = $this->userManager->get($uid);
                                error_log("set email");
                                if (!is_null($user)) {
                                        $user->setEMailAddress($mail);
                                }
                                error_log("set email");
                                return;
                        }
```